### PR TITLE
smarter code gen to not import unneeded types

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] smarter action gen
- [x] smarter engine gen
- [x] facade around errors to not pull in as much

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers

File: protocol/bin/flow.js [hunk 1]
```diff
@@ -386,33 +386,41 @@ function writeActions() {
     disconnected: {},
     connected: {},
   }
+
+  const seenProjects = {}
```
Comment (buoyad) [inline protocol/bin/flow.js]: this can just be the array of project names

File: shared/desktop/yarn-helper/action-creator-creator.tsx [hunk 2]
```diff
@@ -26,12 +26,30 @@ type CompileActionFn = (ns: ActionNS, actionName: ActionName, desc: ActionDesc)
...
+  Object.keys(actions).some(key => payloadHasType(actions[key], toFind))
+
 function compile(ns: ActionNS, {prelude, actions}: FileDesc): string {
+  const immutableImport = actionHasType(actions, /(^|\W)I\./) ? "import * as I from 'immutable'" : ''
+  const rpcGenImport = actionHasType(actions, /(^|\W)RPCTypes\./)
+    ? "import * as RPCTypes from '../constants/types/rpc-gen'"
+    : ''
+
   return `// NOTE: This file is GENERATED from json files in actions/json. Run 'yarn build-actions' to regenerate
 /* eslint-disable no-unused-vars,prettier/prettier,no-use-before-define,import/no-duplicates */
```
Comment (buoyad) [inline shared/desktop/yarn-helper/action-creator-creator.tsx]: can we turn `no-unused-vars` on for these?

Comment (chrisnojima) [reply shared/desktop/yarn-helper/action-creator-creator.tsx]: yup! 🔥

**Pull request merged**